### ISSUE TITLE
Using test mixins disables integration tests

### ISSUE THREAD

**Issue by chmreid** [HumanCellAtlas/data-store#2668]
## Software version (git hash, date, branch)

Commit hash eee134a

Master branch

12/3/2019

## Expected behavior

I expect that if a test is marked as an integration test using the `@testmode.integration` decorator, and the environment variable `DSS_TEST_MODE="integration"`, the integration test will be run.

I also expect that if there are limitations on when `@testmode.integration` can be used (i.e., certain scenarios where the integration test will never be run), it would be documented somewhere, most likely in the file `tests/infra/testmode.py` that defines the integration decorator.

## Actual behavior

If I write a unit test and decorate it with `@testmode.integration`, than let it inherit from all four `TestBundleApi, TestAuthMixin, DSSAssertMixin, DSSUploadMixin` mixin classes, the integration test does not run.

**EDIT:** the problem is only with the `TestBundleApi` class, the other mixins do not break integration tests that inherit from them.

If I define an equivalent test class that does not inherit from any of those four mixin classes, the integration test runs.

## Steps to reproduce the behavior

See branch `chmreid-mixins-break-integration-tests` and associated pull request for an example of a test that reproduces this problem.

To run the integration test that works (no mixins), run `TestFoobar1`:

```
$ DSS_TEST_MODE="integration" python tests/test_foobar.py TestFoobar1
ohai foobar 1 integration test ran okay
.
----------------------------------------------------------------------
Ran 1 test in 0.000s

OK
```

To run the integration test that fails (because of mixins), run `TestFoobar2`:

```
$ DSS_TEST_MODE="integration" python tests/test_foobar.py TestFoobar2
ssssssssssssssssssssssss
----------------------------------------------------------------------
Ran 24 tests in 0.000s

OK (skipped=24)
```

There are also 4 sub-tests that isolate each parent class to see which one breaks the integration tests. It turns out it is `TestBundleApi` class that breaks integration tests, as it is the class associated with `TestFoobar2A`.

```
$ DSS_TEST_MODE="integration" python tests/test_foobar.py TestFoobar2A
ssssssssssssssssssssssss
----------------------------------------------------------------------
Ran 24 tests in 0.000s

OK (skipped=24)

$ DSS_TEST_MODE="integration" python tests/test_foobar.py TestFoobar2B
ohai foobar 2B integration test ran okay
.
----------------------------------------------------------------------
Ran 1 test in 0.000s

OK

$ DSS_TEST_MODE="integration" python tests/test_foobar.py TestFoobar2C
ohai foobar 2C integration test ran okay
.
----------------------------------------------------------------------
Ran 1 test in 0.000s

OK

$ DSS_TEST_MODE="integration" python tests/test_foobar.py TestFoobar2D
ohai foobar 2D integration test ran okay
.
----------------------------------------------------------------------
Ran 1 test in 0.000s

OK
```

## Further notes

* Based on the behavior noted above, the TestOperationsIntegration class in `tests/test_operations.py` will never run, because it is both an integration test and inherits from the mixin classes specified above.

* The order of mixin classes `TestBundleApi, TestAuthMixin, DSSAssertMixin, DSSUploadMixin` can be simplified - `TestAuthMixin` inherits from `DSSAssertMixin`

**Comment by chmreid** [comment]
Oh, the `TestBundleApi` class definition contains a `@testmode.standalone` decorator

https://github.com/HumanCellAtlas/data-store/blob/master/tests/test_bundle.py#L55-L56

**Comment by chmreid** [comment]
This will be closed by #2675
